Dev

LeechFilcher opened the pull request.

Commit:
Fix wierd problems

PKoldborg committed:
Code Simplified & Website UI Scaling Fixed

PKoldborg committed:
About Items back-color more pleasing

PKoldborg committed:
Code Simplified

Commit:
#17 Initial Implementation Content Type Detection

Commit:
#17 Improvements to Content Type Detection and commented public stuff

Commit:
Merge branch 'main' into dev

Commit:
Adjusted Namespaces of Pamaxie.Api

Commit:
#17 Optimized Algorithm and implemented more file Types

Basically optimized the algorithm for more use cases and implemented more file types it supports. Also made it more easily accessible and less demanding. There is still some work to do before its done but most work is there.

PKoldborg committed:
#24 Unit tests for ImageProcessing

Commit:
Update README.md

PKoldborg committed:
#24 Unit test for TokenGenerator.cs

PKoldborg committed:
#24 Reference for Newtonsoft.Json and added appsettings.test.json to the project

PKoldborg committed:
Added appsettings.test.json to .gitignore

PKoldborg committed:
Fixed the wrong path, ref 3c42ef8

LeechFilcher reviewed (commented):
Please mark DownloadFile_Failure as a Theory.

PKoldborg committed:
#34 Moved ImageProcessing.cs to it's own assembly

PKoldborg committed:
#34 Removed unnecessary comment

PKoldborg committed:
#34 Fixed issue where DownloadFile was referring to the path before it got moved.

Commit:
Add a few comments to stuff 

Have  to check this in cause my local repo has issues and I dun wanna loose this work

Commit:
#8 Completion of initial implementation

Initial implementation of Media Detection is complete.

PKoldborg committed:
#13 Progress on Contribution Guideline

PKoldborg committed:
Merge remote-tracking branch 'origin/dev' into dev

PKoldborg committed:
Removed warnings and unnecessary checks for AuthController.cs & VisionController.cs

PKoldborg committed:
Made changes to RefreshTask() in AuthController.cs so it matches LoginTask() more

Commit:
Several Optimizations, excluding Website restructuring

This does not include the restructuring of the website but most other things are included. (Website will be done today or tomorrow)

Commit:
Merge remote-tracking branch 'origin/dev' into dev

Commit:
Fixed API usings

Commit:
Undo Changes

Commit:
Merge remote-tracking branch 'origin/dev' into dev

Commit:
Revert Changes

Commit:
Moved Data into its own assembly and removed unnecessary usings

Commit:
Fixed most warnings except for website

PKoldborg committed:
Auto stash before merge of "dev" and "origin/dev"

PKoldborg committed:
#36 Initial structuralisation of Blazor website

PKoldborg committed:
#6 Website SFG, between Header and MudCarousel, redesigned

PKoldborg committed:
#6 Fixed small style mistake for SVG

PKoldborg committed:
#37 Fixed cookie dialog not showing up, and removed website warnings

PKoldborg committed:
Removed a old folder that should have been deleted some time ago, since the project got moved to another location

PKoldborg committed:
#38 Seperated Sql & Redis from Pamaxie.Database.Extensions into Pamaxie.Database.Extensions.Sql.csproj and Pamaxie.Database.Extensions.Redis.csproj

PKoldborg committed:
Logout now have a returnUrl that can be used & changed some Rider suppress messages to comments as they work the same way.

PKoldborg committed:
Redirect returnUrl needed a base path

PKoldborg committed:
#26 EmailVerificated added to EF database, EmailSender.cs set up, and able to send the user a email, also EmailConfirmation body made (might be changed in the future). UserService.cs is able to show a "Please verify your email" popup when trying to click on 'create application' button.

Commit:
#39 Pamaxie.Tooling - UI Implementation / Rewrite

Rewrote part of pamaxies Tooling with UWP to have a UI. There is probably still bugs in it but its functional so far.

Commit:
Changed workflows so they run on Windows

Commit:
Update codeql-analysis.yml

Commit:
#39 Pamaxie.Tooling - Optimization and implementation of cancellation

Added cancellation and optimized loading as well as saving a bit. The software is currently a bit hard to read this needs to be rewritten. I suspect that this may cause bugs in non optimal scenarios too.

Commit:
Merge remote-tracking branch 'origin/dev' into dev

PKoldborg committed:
Merge remote-tracking branch 'origin/dev' into dev

PKoldborg committed:
#26 Email verification done, and updates the user on the database.

PKoldborg committed:
#26 Token expiration date and a few testing methods for encoding and decoding the token

Commit:
Pamaxie.API - Changed name of Vision Controller and added requests

Commit:
Pamaxie.Data - Resetting the timeout when storing File Hashes in Redis

Commit:
Pamaxie.Clustermanagament - Initial Checkin and Adding the GRPC server

PKoldborg committed:
#26 Email Verification body updated.

PKoldborg committed:
Merge remote-tracking branch 'origin/dev' into dev

PKoldborg committed:
Temporarily build fix

PKoldborg committed:
Optimised Imports

PKoldborg committed:
#26 Email Verification Fix

PKoldborg committed:
#42 added class files for each url and into the dbSets for our SqlDbContext.cs

PKoldborg committed:
#24 Testing structure set up for Pamaxie.Website with mocking.

PKoldborg committed:
Pamaxie.Database.Api.csproj was referring to a assembly that did not exist anymore

PKoldborg committed:
Delete JsonWebTokenTest.cs

Was supposed to be moved, but ended up duplicated.

PKoldborg committed:
Removed appsettings.test.json

PKoldborg committed:
#24 More testing data and easily expandable for Test.Pamaxie.Website

PKoldborg committed:
#24 ClaimExtension.cs accessibility and Misspelled a property name

PKoldborg committed:
Removed some information that will be moved to the wiki

PKoldborg committed:
TestProfileData.cs had the wrong configuration section

PKoldborg committed:
Renamed TestMemberData.cs to MemberData.cs for less confusion.

PKoldborg committed:
#24 Reworked the Testing structure to be less repeatable.

Commit:
Overwork Pamaxie.ImageSorting Tooling into new UI

Commit:
Fixed Overlapping UI Elements

Commit:
Pamaxie.Tooling - Fix Too high accuracy in prediction time.

Commit:
Deleting all kinds of old stuff (This is more than broken.)

Commit:
Breaking even more stuff... like I do

Commit:
#39 Pamaxie.Tooling - Fixed issue with creating a ton of new Models

PKoldborg committed:
#43 Minor fixes to the website that can be done at the moment, and added a interface for PamaxieApplication.cs

PKoldborg committed:
#43 Fixed everything for the testing classes that can be done for now.

PKoldborg committed:
#43 Fixed everything for the testing classes that can be done for now.

PKoldborg committed:
#43 Website errors fixed for the new database structure

PKoldborg committed:
#43 Fixed visual errors in Test.Pamaxie.Website, mocking is still required for some of the tests to work

PKoldborg committed:
Added some summaries for some testing methods that had them missing

PKoldborg committed:
#43 Fixed a few visual errors for Test.Pamaxie.Api, mocking is still needed to be implemented for some methods to work

PKoldborg committed:
#43 made Interactions internal instead of public, to make sure that people uses the extension methods instead

PKoldborg committed:
#43 fixed visual errors for Pamaxie.Api.csproj

PKoldborg committed:
Added a bunch of missing summaries

PKoldborg committed:
Changed method order

PKoldborg committed:
#13 Updated CONTRIBUTING.md

PKoldborg committed:
#13 Updated missing whitespace for sub titles

PKoldborg committed:
#13 Removed repetitive parts and changed text style a bit in a few places.

PKoldborg committed:
#13 Added guideline for documentation

PKoldborg committed:
Updated Packages

PKoldborg committed:
#33 Button rate limit

PKoldborg committed:
Fixed reference error

PKoldborg committed:
#44 cleaned up the code and added comments (#45)

Seems to all work fine.

Commit:
#43 Reworking Server Side (behind API) Database Connection

Commit:
#43 Reworked a bit more stuff relating to server side

PKoldborg committed:
#43 Moved projects to a folder location fitting the directory structure. And added missing implementations for DataServices

PKoldborg committed:
#43 Extension methods for Pamaxie.Database.Extensions.Client

PKoldborg committed:
#43 Adjusted the approach on how the Database.Client will be used

PKoldborg committed:
Visually removed error for UserServiceTest.cs

PKoldborg committed:
#43 Minor adjustments

PKoldborg committed:
#43 Small changes and some dependency fixes

PKoldborg committed:
#43 Added testing projects for the database projects, and added/adjusted some summaries of the other testing projects

PKoldborg committed:
#43 Implemented test classes for Database.Api

PKoldborg committed:
#43 Removed api test calls through httpclient

PKoldborg committed:
#43 Methods ApplicationController.cs and UserController.cs are created with unit tests.

PKoldborg committed:
#43 Test for TokenGenerator.cs and minor adjustments

PKoldborg committed:
#43 Redesigned testing structure a bit

PKoldborg committed:
Added summaries for BaseTest.cs

PKoldborg committed:
#43 Moved test instance and password to appsettings.test.json

PKoldborg committed:
Removed a testing method, that was not supposed to be committed

PKoldborg committed:
#43 Mocking for UserDataService.cs and ApplicationDataService.cs, changed summaries to make more sense for our design

PKoldborg committed:
#43 Reformatted code for Testing projects

PKoldborg committed:
#43 Mocked Application Create, adds the key to the owner's ApplicationKeys list & mocked VerifyAuthentication implemented

PKoldborg committed:
Minor changes to summaries

PKoldborg committed:
This newline triggered me

PKoldborg committed:
#43 Moved DataService mock calls to constructor

PKoldborg committed:
#43 Test class for MockUserDataService.cs

Still in progress

PKoldborg committed:
#43 Test for all mocked DataService interactions and added GetOwner for application

PKoldborg committed:
Removed AuthorizationTokenCipher from application testing data.

PKoldborg committed:
Removed BCryptExtension.cs

PKoldborg committed:
#43 Disabled Test Parallelization

PKoldborg committed:
Removed newline that accidentally made it into 6f1a35

PKoldborg commented:
Once database_rework have been merged to dev, and everything have been tested and works, then dev will merge into main.

PKoldborg committed:
Removed Base from file names

PKoldborg committed:
Added GetOwner to ApplicationDataService.cs in .Server

PKoldborg committed:
#43 Removed dependency for .Server in Test.Base.csproj, uses the interface instead

Commit:
#53 - Added methods to get the user Id from the bearer token

PKoldborg committed:
Test fix for b8bf940

Commit:
#43 - Changed Folder structure and optimized some code.

PKoldborg committed:
Merge remote-tracking branch 'origin/database-rework' into database-rework

Commit:
Reset Commit to fix issues

Commit:
Merge remote-tracking branch 'origin/database-rework' into database-rework

Commit:
Added code style

PKoldborg committed:
#43 Minor Changes

PKoldborg committed:
Merge remote-tracking branch 'origin/database-rework' into database-rework

Commit:
Fixed cause cold was annoyed at me for putting it somewhere wrong :(

Commit:
Auto stash before merge of "database-rework" and "origin/database-rework"

PKoldborg committed:
#43 fixed some minor problems with 8d8fbe7

Commit:
#43 Fixed some stupid dependency injections, updated assemblies.

PKoldborg committed:
#43 Fixed a issue where a service could not be constructed

PKoldborg committed:
Merge remote-tracking branch 'origin/database-rework' into database-rework

PKoldborg committed:
#43 Small fixes

Commit:
#43 Created worker services for distribution

Commit:
#40 Create projects for analysis (WorkerServices)

Commit:
Remove unused files / folders

Commit:
Moved more of the folders around to fix the folder structure

PKoldborg committed:
#43 Fixed dependency issues, removed unused NuGet packages, fixed warnings, cleaned up a bit code & added test project for Pamaxie.WorkerService.Api.csproj & Pamaxie.Jwt.csproj

PKoldborg committed:
#43 Changed testing namespaces, added some testing projects and classes, added testing methods and data for ImageProcessing.cs & fixed a warning in NavMenu.razor

PKoldborg committed:
#43 added missing NuGet Package for running tests

PKoldborg committed:
#43 Fixed Stream.Position = 0 throwing 'Specified argument was out of the range of valid values.'

PKoldborg committed:
#43 added a few FileTypes, Test data for ImageProcessingTest.cs and fixed a issue in ImageProcessing.cs

PKoldborg committed:
#43 Testing for FileDetection.cs

PKoldborg committed:
Small changes

PKoldborg committed:
Reformatted and cleaned up code

PKoldborg committed:
#43 Testing fully done for all Assembly projects, and fixed a issue in TokenGenerator.cs where it would not return the correct claim

PKoldborg committed:
#43 Sealed most classes

PKoldborg committed:
#43 Changed T from being IPamaxieUser/IPamaxieApplication to be PamaxieUser/PamaxieApplication and added Redis database mocking for Database.Api testing

PKoldborg committed:
#43 Some testing progress on Database.Api

PKoldborg committed:
#43 Tests for Database.Server & Database.Client, fixed a lot of other tests and small stuff in other projects

PKoldborg committed:
#43 Some code cleanup

PKoldborg committed:
#52 Reworked ApplicationController.cs and added Swagger for easy testing when debugging. Tests are broken and will be fixed another day.

PKoldborg committed:
#52 Reworked the database controllers to use proper status codes

Testing classes still needs to be reworked, but will not be done for this issue, as all testing projects will be changed to fit the style guide

PKoldborg committed:
#52 Small fixes

PKoldborg committed:
#43 Changed MemberData.cs and changed all tests for Pamaxie.Database.Api

PKoldborg committed:
#43 All test projects reworked to fit current style guide

PKoldborg committed:
#43 Changed ImageProcessing, Jwt, MediaDetection, Database.Client, Database.Server and .Api for the new style guideline & minor fixes for testing methods.

PKoldborg committed:
#43 Added Status404NotFound for HttpGet if the key does not exist

PKoldborg committed:
#43 Database.Client implemented with test classes & added a missing controller method for Database.Api

PKoldborg committed:
Minor cleanup

PKoldborg committed:
#43 Minor test fixes

Commit:
#43 Optimizations in the Client and Api

PKoldborg committed:
#43 Minor cleanup

PKoldborg committed:
#43 fixed Database.Client test methods, worked on some TODOs & moved some setting validation/generation into it's own helper project

PKoldborg committed:
#43 Split generation sections up, added a bit more validation for SettingsValidation.cs & SettingsGeneration.cs can now save the configuration file

PKoldborg committed:
#43 Small friday fixes and changes

Commit:
#43 Small Friday improvements and changes

PKoldborg committed:
#43 Moved ServerBase.cs and ApiBase.cs out of Test.Base.csproj to prevent all test projects to have access to .Client and .Server, changed how the Database.Server and Database.Client will handle DataContext

PKoldborg committed:
Merge remote-tracking branch 'origin/main' into database-rework

PKoldborg committed:
#54 appsettings.test.json removed and configuration is now generated

PKoldborg committed:
Some summary updates

Commit:
Merge remote-tracking branch 'origin/database-rework' into database-rework

Commit:
Auto stash before merge of "database-rework" and "origin/database-rework"

Commit:
#43 Fixed merge Conflicts and errors

Commit:
Testing if commits work again

Commit:
Test if verification works now

Commit:
#43 Finished Design on Server Side implementation

Commit:
#43 Finished work on the configuration and started working on the API

Commit:
Finished work on the configuration and started working on the API

Commit:
Merge remote-tracking branch 'origin/database-rework' into database-rework

Commit:
#43 smaller fixes to auth and db design

Commit:
Updated the Token Generation Logic

Commit:
Fixed Authentication for Database API

PKoldborg committed:
Merge branch 'main' into dev

# Conflicts:
#	.gitignore

PKoldborg committed:
Merge branch 'dev' into database-rework

# Conflicts:
#	.gitignore
#	Pamaxie.Website/Pamaxie.Website.csproj
#	Pamaxie.Website/Startup.cs
#	Pamaxie.sln
#	Tooling/Pamaxie.ImageTooling/App.xaml
#	Tooling/Pamaxie.ImageTooling/Assets/Big.png
#	Tooling/Pamaxie.ImageTooling/MainWindow.xaml
#	Tooling/Pamaxie.ImageTooling/MainWindow.xaml.cs
#	Tooling/Pamaxie.ImageTooling/Pamaxie.ImageTooling.csproj
#	Tooling/Pamaxie.ImageTooling/Pamaxie.ico
#	Tooling/Pamaxie.ImageTooling/Pamaxie.snk
#	Tooling/Pamaxie.ImageTooling/PresentationObjects/PoComparisonSettings.cs
#	Tooling/Pamaxie.ImageTooling/PresentationObjects/PoImageData.cs
#	Tooling/Pamaxie.ImageTooling/PresentationObjects/PoProcessingSettings.cs
#	Tooling/Pamaxie.ImageTooling/Properties/Resources.Designer.cs
#	Tooling/Pamaxie.ImageTooling/Properties/Resources.resx
#	Tooling/Pamaxie.ImageTooling/Properties/Settings.Designer.cs
#	Tooling/Pamaxie.ImageTooling/Properties/Settings.settings
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImageComparisonViewModel/ImageComparisonViewModel.Fields.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImageComparisonViewModel/ImageComparisonViewModel.Methods.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImageComparisonViewModel/ImageComparisonViewModel.Properties.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImageComparisonViewModel/ImageComparisonViewModel.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImagePreparationViewModel/ImagePreparationViewModel.Fields.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImagePreparationViewModel/ImagePreparationViewModel.Methods.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImagePreparationViewModel/ImagePreparationViewModel.Properties.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/ImagePreparationViewModel/ImagePreparationViewModel.cs
#	Tooling/Pamaxie.ImageTooling/ViewModels/MainViewModel/MainViewModel.Methods.cs
#	Tooling/Pamaxie.ImageTooling/appsettings.json

PKoldborg committed:
Merge fix

PKoldborg committed:
Merge pull request #57 from pamaxie/database-rework

Database rework to Dev

Commit:
Partial Checkin (#58)

Commit:
Fixed a bug in the authentication of the Database API

Commit:
Started implementation of C# Client.

Commit:
Checkin, fixed a bunch of errors. Removed things that aren't needed

Commit:
Fix workflow

Commit:
Fixed Warnings and some Build Errors

Commit:
Update dotnet.yml
